Remora.Rest/Json/DataObjectConverter.cs

```diff
@@ -600,30 +621,13 @@ JsonSerializerOptions options
 
         foreach (var dtoProperty in _dtoProperties)
         {
-            var propertyGetter = dtoProperty.GetGetMethod();
-            if (propertyGetter is null)
-            {
-                continue;
-            }
-
             if (!dtoProperty.CanWrite && !_includeReadOnlyOverrides.Contains(dtoProperty))
             {
                 // Skip read-only properties, unless told otherwise.
                 continue;
             }
 
-            // The value might be some sort of overriding type with explicit implementations; we'll map over to the
-            // interface if necessary
-            if (value.GetType() != typeof(TImplementation))
```

Comment:
I don't think there are test cases for this specifically. I think crafting a special test case for this would make sense, but there's also no test project in Remora.Rest; would you like me to create one?

Reply:
Right here :) https://github.com/Nihlus/Remora.Rest/tree/master/Tests/Remora.Rest.Tests